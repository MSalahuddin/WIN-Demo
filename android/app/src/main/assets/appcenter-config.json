{
    "app_secret":"$process.env.APP_CENTER_ID"
}