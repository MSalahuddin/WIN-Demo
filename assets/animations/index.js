/* eslint-disable global-require */

export const redBlueArrowAnimation = require('./red_blue_arrow.json');
export const redYellowArrowAnimation = require('./red_yellow_arrow.json');
export const redYellowCircleAnimation = require('./red_yellow_circle.json');
export const starsAnimation = require('./stars_with_bg.json');
export const gamePlayAnimation = require('./claw.json');
export const quickRefill = require('./quick_refill.json');
export const congratsYouWonAnimation = require('./congrats_you_won.json');
/* eslint-enable global-require */
