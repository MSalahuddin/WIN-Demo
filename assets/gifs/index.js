/* eslint-disable global-require */

export const gamePlayLoading = require('./game_play_loading.gif');

/* eslint-enable global-require */
