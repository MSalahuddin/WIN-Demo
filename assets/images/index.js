/* eslint-disable global-require */
export const arrowBack = require('./arrow_back.png');
export const arrow = require('./arrow.png');
export const checkBoxActive = require('./checkbox_active.png');
export const howtoBackgroundPopUp = require('./how_to_background.png');
export const howToIcon = require('./how_to_icon.png');
export const wwIcon = require('./ww_icon.png');
export const tokensToPlayBackground = require('./tokens_to_play_background.png');
export const offerwallIcon = require('./action.png');
export const offerwallBackround = require('./offerwall_background.png');
export const crossNotification = require('./cross_notification.png');
export const notificationBell = require('./neon_notification_icon.png');
export const trackingIcon = require('./tracking_icon.png');
export const empty = require('./icon_empty.png');
export const refill = require('./popup_refill.png');
export const full = require('./icon_full.png');
export const piggyBankBlueCircle = require('./piggy_bank_blue_circle.png');
export const piggyBank = require('./piggy_bank.png');
export const piggyBankBonus = require('./bonus_piggy_bank.png');
export const bonusStars = require('./bonus_stars.png');
export const multiChicken = require('./multi_chicken.png');
export const BonusCoin = require('./dailybonus_coin.png');
export const piggyBankBallon = require('./piggy_bank_balloon_banner.png');
export const purpleBackgroundStars = require('./purple-background-stars.png');
export const iconRaf = require('./icon_raf.png');
export const wwVipChicken = require('./ww_vip_chicken.png');
export const facebookIcon = require('./facebook_icon.png');
export const twitterIcon = require('./twitter_icon.png');
export const smsIcon = require('./sms_icon.png');
export const moreIcon = require('./more.png');
export const emailIcon = require('./email.png');
export const cloudyBackground = require('./cloudy_background.png');
export const dealOfTheDayNoneVip = require('./none-vip-dod.png');
export const dealOfTheDayVip = require('./vip-dod.png');
export const dailyBonusCoin = require('./daily_bonus_coin.png');
export const apple = require('./apple.png');
export const arrowInCircle = require('./arrow_in_circle.png');
export const backInactive = require('./back_inactive.png');
export const back = require('./back.png');
export const backgroundStars = require('./background_stars.png');
export const balloonBow = require('./balloon_bow.png');
export const balloonFloat = require('./balloon_float.png');
export const banner = require('./banner.png');
export const blackBoxYellowDots = require('./black_box_yellow_dots.png');
export const bannerBlue = require('./banner_blue.png');
export const bannerLost = require('./banner_lost.png');
export const bannerCoin = require('./coin_banner.png');
export const bannerPurple = require('./banner_purple.png');
export const bannerPurpleBig = require('./banner_purple_big.png');
export const bannerSale = require('./banner_sale.png');
export const bannerWithBowTie = require('./banner_bow_tie.png');
export const bannerWithStar = require('./banner_with_star.png');
export const bannerWon = require('./banner_won.png');
export const bannerExited = require('./Banner_exited.png');
export const calendar = require('./calendar.png');
export const cameraButton = require('./camera_button.png');
export const checkedBox = require('./checked_box.png');
export const chickenTux = require('./chicken_tux.png');
export const chickenWithBalloon = require('./chicken_with_balloon.png');
export const chickenWithPhone = require('./chicken_with_phone.png');
export const clock = require('./clock.png');
export const closeBlack = require('./close_black.png');
export const closeWhite = require('./close_white.png');
export const closeCta = require('./cta-close.png');
export const coin = require('./coin.png');
export const coinBucket = require('./coin_bucket.png');
export const controlPanelBackground = require('./control_panel_background.png');
export const countdownOne = require('./countdown_one.png');
export const countdownThree = require('./countdown_three.png');
export const countdownTwo = require('./countdown_two.png');
export const crane = require('./white_claw.png');
export const downArrow = require('./down_arrow.png');
export const downButtonActive = require('./down_active.png');
export const downButtonInctive = require('./down_inactive.png');
export const error = require('./error.png');
export const eye = require('./eye.png');
export const facebook = require('./facebook.png');
export const fallingChicken = require('./falling-chicken.png');
export const ferrisWheelCart = require('./ferris_wheel_cart.png');
export const gear = require('./gear.png');
export const gearBorderless = require('./gear_borderless.png');
export const giftBox = require('./giftbox.png');
export const giftGray = require('./gift_gray.png');
export const giftMedal = require('./gift_medal.png');
export const goldBars = require('./gold_bars.png');
export const goldStar = require('./gold_star.png');
export const goldBarsPlus = require('./gold_bars_plus.png');
export const google = require('./google.png');
export const grabMainButton = require('./grab.png');
export const greenArrow = require('./green_arrow.png');
export const lastWinArrow = require('./last_win_arrow.png');
export const greenCircleArrow = require('./green_circle_arrow.png');
export const greenLight = require('./green_light.png');
export const greyTicket = require('./grey_ticket.png');
export const happyChicken = require('./happy_chicken.png');
export const hourGlass = require('./hourglass.png');
export const info = require('./info.png');
export const gameInfo = require('./icon_info.png');
export const leftButtonActive = require('./left_active.png');
export const leftButtonInctive = require('./left_inactive.png');
export const line = require('./line.png');
export const mainButtonInactive = require('./main_inactive.png');
export const mascotPartyHat = require('./mascot_party_hat.png');
export const medalClock = require('./medal_clock.png');
export const medalCoin = require('./medal_coin.png');
export const medalError = require('./medal_error.png');
export const medalGold = require('./medal_gold.png');
export const medalHourGlass = require('./medal_hour_glass.png');
export const medalInfo = require('./medal_info.png');
export const medalStar = require('./medal_star.png');
export const onboardingBecomeVIP = require('./onboarding_become_vip.png');
export const onboardingGetNews = require('./onboarding_get_news.png');
export const onboardingWelcome = require('./onboarding_welcome.png');
export const onboardingWinRealPrizes = require('./onboarding_win_real_prizes.png');
export const oval = require('./oval.png');
export const paginationDotActive = require('./pagination_active.png');
export const paginationDotInactive = require('./pagination_inactive.png');
export const playMainButton = require('./play.png');
export const prize = require('./prize.png');
export const purpleBoxYellowDots = require('./purple_box_yellow_dots.png');
export const purseCoins = require('./purse_coins.png');
export const purseCoinsMedal = require('./purse_coins_medal.png');
export const redBack = require('./red_back.png');
export const redLight = require('./red_light.png');
export const rightButtonActive = require('./right_active.png');
export const rightButtonInctive = require('./right_inactive.png');
export const sadChicken = require('./sad_chicken.png');
export const saleBanner = require('./sale_banner.png');
export const share = require('./share.png');
export const star = require('./star.png');
export const tent = require('./tent.png');
export const tick = require('./tick.png');
export const ticket = require('./ticket.png');
export const triangle = require('./triangle.png');
export const upButtonActive = require('./up_active.png');
export const upButtonInctive = require('./up_inactive.png');
export const vipBanner = require('./vip_banner.png');
export const gameVipBanner = require('./game_banner_vip.png');
export const watchIcon = require('./watch.png');
export const watchGreen = require('./watch_green.png');
export const winnerLogo = require('./winner_logo.png');
export const winningBanner = require('./winning_banner.png');
export const mobileChicken = require('./mobile_chicken.png');
export const backgroundVip = require('./background_vip.png');
export const search = require('./search_icon.png');
export const searchIcon = require('./search.png');
export const allMachinesIcon = require('./all_machines_icon.png');
export const selectDifficulty = require('./select_difficulty_icon.png');
export const controlSearch = require('./controls_search.png');
export const notificationOn = require('./notification_on.png');
export const plus = require('./plus.png');
export const prizeWon = require('./leaderboard.png');
export const giftBoxTransparent = require('./prize_vault.png');
export const piggyBankIcon = require('./piggy_bank_icon.png');
export const notification = require('./news.png');
export const rewardedVideo = require('./popup_rewarded_video_circle.png');
export const rewardedVideoBonus = require('./bonus_rewarded_video.png');
export const CenterCircleBackGround = require('./centered_circles_background.png');
export const ScatteredCircleBackGround = require('./scattered_circles_background.png');
export const LostClaw = require('./icon_loss_claw.png');
export const ExitCircle = require('./exit-circle.png');
export const preLoad = require('./pre-load.png');
export const postLoad = require('./post-load.png');
export const onLoad = require('./on-load.png');
export const LeaderboardRows = require('./leaderboard-rows.png');
export const CategoryMyRank = require('./category-my-rank.png');
export const CategoryMyWins = require('./category-wins.png');
export const AlternateImage = require('./userImage1.png');
export const MyWinImage = require('./wins-icon.png');
export const MyRankImage = require('./ranking_icon.png');
export const bottomNavCross = require('./cross_white.png');
export const BgImageDailyBonus = require('./daily_bonus_background.png');
export const ImageIconDailyBonus = require('./daily_bonus.png');
export const BgImageReferFriend = require('./refer_a_friend_background.png');
export const ImageIconReferFriend = require('./raf.png');
export const BgImagePiggyBank = require('./piggy_bank_background.png');
export const ImageIconPiggyBank = require('./piggybank.png');
export const BgImageStackEm = require('./stack_em_background.png');
export const ImageIconStackEm = require('./stackEm.png');
export const BgImageWatchAndEarn = require('./watch_and_earn_background.png');
export const ImageIconWatchAndEarn = require('./watch_videos.png');
export const BgImageWinTen = require('./win_10_background.png');
export const ImageIconWinTen = require('./win10.png');
export const BgImageWinFifty = require('./50_in_a_week_background.png');
export const ImageIconWinFifty = require('./win50.png');
export const refferFriendIcon = require('./refer_a_friend_icon.png');
export const inviteFriendBottomCoins = require('./bottom_tokens.png');
export const inviteFriendBackgroundTokenImage1 = require('./bg_invite_token1.png');
export const inviteFriendBackgroundTokenImage2 = require('./token_03.png');
export const upRollToken = require('./token_04.png');
export const downToken = require('./token_05.png');
export const piggyBankHeader = require('./piggy_bank_header.png');
export const backGroundPiggyBank = require('./background_piggy_bank.png');
export const backGroundDealOfTheDay = require('./deal_of_the_day_token_popup_background.png');
export const DownTokenRight = require('./deal_of_the_day_token_05.png');
export const DailyBonusHeader = require('./daily_bonus_header.png');
export const DealoftheDayImage = require('./deal_of_the_day.png');
export const ImageIconFreePlays = require('./free_icon_0.png');
export const BgImageFreePlays = require('./free_plays_background.png');
export const NotificationCircle = require('./bubble_amount.png');
export const FreePlayPopUpButton = require('./button_us_a_free_play.png');
export const FreeToken = require('./icon-free-tokens.png');
export const LastWinIcon = require('./icon-last-win.png');
export const BlankBanner = require('./blank.png');
export const StarBanner = require('./stars.png');
export const SubscriptionUperBannerBackground = require('./upper_box_background.png');
export const SubscriptionLowerBannerBackground = require('./lower_box_background.png');
export const LostTag = require('./lost.png');
export const WonTag = require('./won.png');
export const BlueButton = require('./blue.png');
export const GrayButton = require('./gray.png');
export const iconFailure = require('./icon_failure.png');
export const iconTokenGrant = require('./icon_token_grant.png');
export const iconProfile = require('./icon_profile.png');
export const onBoardingBackgroundImage = require('./Standard.png');
export const eyeWatchImage = require('./icon_watch_others.png');
export const realMachineImage = require('./icon_real_machines.png');
export const realPrizeImage = require('./icon_real_prizes.png');
export const lockAccessImage = require('./icon_access.png');
export const prizeIconImage = require('./icon_big_prizes.png');
export const freeShippingImage = require('./icon_free_shipping.png');
export const winRealPrizesBgImage = require('./Win_Real_Prizes_Background.png');
export const NotificationsBgImage = require('./Notifications_Background.png');
export const NotificationIcon = require('./icon_notifications.png');
export const HelperSwipeUpImage = require('./helper-swipe-up.png');
export const HelperSearchYourFav = require('./helper-search-your-favorites.png');
export const helperPlaySelectGames = require('./play_select_free_games.png');
export const helperWinnerCrile = require('./helper-winners-circle.png');
export const helperNeedTokens = require('./token_store_helper_copy.png');
export const helperYourPrizeVault = require('./helper-your-prize-vault.png');
export const helperShipOrSwap = require('./helper-ship-or-swap.png');
export const iconShare = require('./share_icon.png');
export const iconFavoriteSelected = require('./like_icon_selected.png');
export const iconFavoriteUnselected = require('./like_icon_unselected.png');
export const iconGameLevelBeginner = require('./Beginner.png');
export const iconGameLevelIntermediate = require('./intermediate.png');
export const iconGameLevelExpert = require('./expert.png');
export const iconGameLevelPro = require('./pro.png');
export const checkBack = require('./check_back_soon.png');
export const iconLikeSelected = require('./icon_like_selected.png');
export const iconLikeUnselected = require('./icon_like_unselected.png');
export const iconPlayWinVideo = require('./icon_play_win_video.png');
export const bannerNewNews = require('./banner_new_news.png');
export const plusGreenIcon = require('./plus_green.png');
export const categoriesIcon = require('./icon_categories.png');
export const topPlayerIcon = require('./icon_topplayers.png');
export const iconSearch = require('./icon_search.png');
export const challengeIcon = require('./challenges.png');
export const upArrows = require('./up_arrows.png');
export const downArrows = require('./down_arrows.png');
export const collection = require('./collections.png');
export const todays_challenge = require('./todays_challenge.png');
export const weekly_challenge = require('./weekly_challenge.png');
export const monthly_challenge = require('./monthly_challenge.png');
export const back_arrow = require('./back_arrow.png');
export const challenge_description = require('./challenge_description.png');
export const challenge_ticket = require('./challenge_tickets.png');
export const challenge_token = require('./challenge_tokens.png');
export const challenge_rank = require('./challenge_rank.png');
export const challenge_grand_prize = require('./challenge_grand_prize.png');
export const challenge_3rd_place = require('./3rd_place.png');
export const challenge_2nd_place = require('./2nd_place.png');
export const play_history_share = require('./play_history_share.png');
export const play_history_token = require('./play_history_token.png');
export const play_history_watch = require('./play_history_watch.png');
export const report_flag = require('./report_flag.png');
export const reported_flag = require('./reported_flag.png');
export const watch_ad = require('./watch_ad.png');
export const rank_view_all = require('./rank_view_all.png');
export const header_top_players = require('./header_top_players.png');
export const menu_bar = require("./menu.png");
export const more_icon = require("./more_icon.png")
export const email_icon = require("./email_icon.png")
export const raf_icon = require("./raf_icon.png")
export const myWalletBackButton = require("./myWalletBackButton.png")
export const myTokensIcon = require("./myTokensIcon.png")
export const QRImage = require("./qr-code-scan.png")
export const plusTokens = require("./plus_tokens.png")
export const ticketIcon = require("./ticketicon.png");
export const newSaleBanner = require("./Sale.png");
export const nftPopupIcon = require("./nft_popup_icon.png");
export const confirmPopupIcon = require("./confirm_popup_icon.png");
export const transferredPopupIcon = require("./transferred_popup_icon.png");
export const plusTokensNew = require("./plus_tokens_new.png");
export const successful = require("./successful.png");
export const unavailable = require("./unavailable.png");


export const ww_logo_white = require('./ww_logo_white.png');
export const players_watching = require('./players_watching.png');
export const share_button = require('./share_button.png');
export const comment = require('./comment.png');
export const play_button = require('./play_button.png');
export const join = require('./join.png');
export const eliminated = require('./eliminated.png');
export const new_life = require('./new_life.png');
export const keep_playing = require('./keep_playing.png');
export const notification_icon = require('./notification_icon.png');
export const question_icon = require('./question_icon.png');
export const active_players_icon = require('./active_players_icon.png');
export const trivia_background = require('./trivia_background.png');
export const trivia_logo = require('./trivia_logo.png');
export const rewardsIcon = require('./rewards_icon.png');
export const newCategories = require('./newCategories.png');
export const newExplore = require('./newExplore.png');
export const newLeaderboard = require('./newLeaderboard.png');
export const newSearchGameroom = require('./newSearchGameroom.png');
export const newSearch = require('./new-search.png');
export const prizeCreation = require('./prize-creation.png');
export const winnerCircleCheck = require('./check.png');
export const newHeaderTickket = require('./new-header-tickket.png');
export const newWinnerCircleSaleBanner = require('./newWinnerCircleSaleBanner.png');
export const crossChosenCategory = require('./crossChosenCategory.png');
export const triviaNewBackground = require('./trivia_new_background.png');
export const liveTriviaLogo = require('./live_trivia_logo.png');
/* eslint-enable global-require */
