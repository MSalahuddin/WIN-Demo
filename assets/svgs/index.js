import Win_demoWIcon from './Win_demo_w.svg';

export { 
    Win_demoWIcon,
};
