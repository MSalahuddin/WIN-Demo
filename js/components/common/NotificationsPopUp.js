import React, { useEffect, useState } from 'react';
import { Platform } from 'react-native';
import { checkNotifications, openSettings, requestNotifications } from 'react-native-permissions';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import Text, { SIZE, FONT_FAMILY } from './Text';
import Button, { BUTTON_COLOR_SCHEME } from './Button';
import { scaleHeight } from '../../platformUtils';
import { color } from '../../styles';
import { bannerString } from '../../stringConstants';
import { notificationBell, crossNotification } from '../../../assets/images';



const Backdrop = styled.View`
  flex-direction: row;
  background-color: ${color.iceBlue};
  align-self: center;
  justify-content: center;
  align-items: center;
  width: 90%;
  border-radius: 10;
  margin-top: 10
  padding-vertical: 5
`;

const TextContainer = styled.View`
  align-items: flex-start
  width: 50%;
  border-radius: 5;
`;

const ButtonContainer = styled.View`
  align-items: center;
  justify-content: center;
  width: 25%;
`;

const Image = styled.Image`
  width: 70;
  height: 70;
  margin-bottom: 10
`;

const ImageContainer = styled.View`
  align-items: center;
  justify-content: center;
  width: 25%;

`;

const ButtonText = styled(Text)`
  margin-bottom: ${Platform.OS === 'android' ? 5 : 0};
`;

 
const ButtonWrapper = styled(Button)`
  margin-vertical: ${scaleHeight(20)};
`;

const ButtonContentContainer = styled.View`
  align-items: center;
  flex-direction: row;
  justify-content: center;
`;

const CancelButtonContainer = styled.TouchableOpacity`
  align-items: center;
  justify-content: center;
  position: absolute
  top: ${({ top }) => (top + 3)} ;
  left: ${({ left }) => (left - 10)} ;
  z-index: 1;

`;

const CrossImage = styled.Image`
  width: 20;
  height: 20;
`;


const CancelButton = ({ onPress, coordinates }) => {

  if(coordinates){
    return (
      <CancelButtonContainer
      top={coordinates.y}
      left={coordinates.width}
      onPress={onPress}
      >
        <CrossImage source={crossNotification}/>
      </CancelButtonContainer>
    );
  }
  return null
  
};
CancelButton.propTypes = {
  coordinates: PropTypes.shape({
    y: PropTypes.number.isRequired,
    width: PropTypes.number.isRequired,
  }).isRequired,
  onPress: PropTypes.func.isRequired,
};

const NotificationsPopUp = () => {

  const [visible, setVisible] = useState(false);
  const [coordinates, setCoordinates] = useState(null);


  const isViewPushNotificationPrompt = async () => {
    const permission = await checkNotifications()
    if (permission.status === 'denied' || permission.status === 'blocked') {
      setVisible(true)
    }
  }

  useEffect(()=>{
    isViewPushNotificationPrompt()
  },[])

  const onPress = async() => {
    setVisible(false)
    const permissionNotifications = await checkNotifications()
    if (Platform.OS === 'ios') {
      if (permissionNotifications.status === 'denied') {
        await requestNotifications(['alert', 'sound', 'badge'])
      }
    }
    if (permissionNotifications.status === 'blocked') {
        await openSettings()
    }
  }

  const onCross = async () => {
    setVisible(false)
  }
    if(visible){
      return (
      <>
        <Backdrop
        onLayout={event => {
          setCoordinates(event.nativeEvent.layout)
        }} 
        >
            <ImageContainer>
              <Image source={notificationBell}/>
            </ImageContainer>
            <TextContainer>
              <Text alignCenter size={SIZE.SMALL} fontFamily={FONT_FAMILY.BOLD} color={color.white}>
                {bannerString.turnOnNotifications}
              </Text>
              <Text alignCenter size={SIZE.XXSMALL} fontFamily={FONT_FAMILY.REGULAR} color={color.white}>
                {bannerString.notificationBannerLineOne}
              </Text>
              <Text alignCenter size={SIZE.XXSMALL} fontFamily={FONT_FAMILY.REGULAR} color={color.white}>
                {bannerString.notificationBannerLineTwo}
              </Text>
            </TextContainer>
            <ButtonContainer>
            <ButtonWrapper
              borderRadius={44}
              height={35}
              width={70}
              theme={BUTTON_COLOR_SCHEME.GREEN}
              onPress={onPress}
            >
              <ButtonContentContainer>
                <ButtonText color={color.white} size={SIZE.XSMALL}>
                {bannerString.enable}
                </ButtonText>
              </ButtonContentContainer>
            </ButtonWrapper>
            </ButtonContainer>
        </Backdrop>
        <CancelButton coordinates={coordinates} onPress={onCross} />  
        </>
    );
    }
    
      return null
    
  
  
};



export default NotificationsPopUp;
