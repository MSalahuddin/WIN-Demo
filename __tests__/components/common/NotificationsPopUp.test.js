import React from 'react';
import { render } from 'react-native-testing-library';
import MockProvider from '../../MockProvider';
import NotificationsPopUp from '../../../js/components/common/NotificationsPopUp';
import { mobileChicken } from '../../../assets/images';

describe('Component NotificationsPopUp', () => {
  const onPress = jest.fn();
  const secondaryButtonOnPress = jest.fn();

  test('show NotificationsPopUp', () => {
    const output = render(
      <MockProvider>
        <NotificationsPopUp
          buttonText="a"
          bannerLabel="a"
          backdropText="a"
          icon={mobileChicken}
          isVisible
          onPress={onPress}
          secondaryButtonOnPress={secondaryButtonOnPress}
        />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });
});
