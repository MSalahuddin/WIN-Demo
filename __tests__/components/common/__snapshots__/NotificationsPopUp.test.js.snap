// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component NotificationsPopUp show NotificationsPopUp 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(34,34,34, 0.6)",
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 37.5,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "transform": Array [
            Object {
              "translateY": 1334,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#A5C1FF",
              "borderColor": "#FFFFFF",
              "borderRadius": 88,
              "borderWidth": 3,
              "height": 176,
              "justifyContent": "center",
              "marginTop": 236,
              "width": 176,
            },
          ]
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 136,
                "width": 136,
              },
            ]
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "marginTop": 48,
            },
          ]
        }
        width={686}
      >
        <Image
          bannerType="NORMAL"
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 131.23478260869567,
                "width": 686,
              },
            ]
          }
          width={686}
        />
        <Text
          alignCenter={false}
          allowFontScaling={false}
          bannerType="NORMAL"
          color="#000000"
          fontFamily="BOLD"
          fontSize={22}
          isUnderlined={false}
          isUppercase={true}
          maxFontSizeMultiplier={1.3}
          size="LARGE"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "CalibreBold",
                "fontSize": 31.5,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#000000",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "uppercase",
              },
              Object {
                "color": "#FFFFFF",
                "letterSpacing": 1.5,
                "position": "absolute",
              },
            ]
          }
        >
          a
        </Text>
      </View>
      <View
        hasChildren={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderBottomLeftRadius": 16,
              "borderBottomRightRadius": 16,
              "height": 306,
              "justifyContent": "flex-start",
              "marginTop": -40,
              "paddingTop": 40,
              "shadowColor": "rgba(53, 53, 53, 0.3)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 2,
              "width": "80%",
              "zIndex": -1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "marginHorizontal": 15,
                "marginTop": 30,
              },
            ]
          }
        >
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#353535"
            fontFamily="REGULAR"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="XSMALL"
            style={
              Array [
                Object {
                  "color": "#353535",
                  "fontFamily": "CalibreRegular",
                  "fontSize": 24,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#353535",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
              ]
            }
          >
            a
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 10,
              "marginVertical": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(96,202,142,1)",
              "borderRadius": 88,
              "height": 176,
              "justifyContent": "center",
              "marginTop": -92,
              "opacity": 1,
              "width": 522,
            }
          }
          testID="popup-button"
        >
          <View
            borderRadius={44}
            colorScheme="rgba(128,229,173,1)"
            height={88}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(128,229,173,1)",
                  "borderRadius": 88,
                  "height": 166,
                  "justifyContent": "center",
                  "width": 516,
                },
              ]
            }
            width={261}
          >
            <View
              borderRadius={44}
              colorScheme="rgba(114,213,158,1)"
              height={88}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(114,213,158,1)",
                    "borderRadius": 108,
                    "height": 162,
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": 516,
                  },
                ]
              }
              width={261}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  alignCenter={false}
                  allowFontScaling={true}
                  color="#FFFFFF"
                  fontFamily="MEDIUM"
                  isUnderlined={false}
                  isUppercase={false}
                  maxFontSizeMultiplier={1.3}
                  size="XLARGE"
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "Calibre-Medium",
                        "fontSize": 36,
                        "justifyContent": "center",
                        "textAlign": "left",
                        "textDecorationColor": "#FFFFFF",
                        "textDecorationLine": "none",
                        "textDecorationStyle": "solid",
                        "textTransform": "none",
                      },
                      Object {
                        "marginTop": 14,
                      },
                    ]
                  }
                >
                  a
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#FFFFFF",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 48,
            "opacity": 1,
          }
        }
      >
        <Text
          alignCenter={false}
          allowFontScaling={true}
          color="#FFFFFF"
          fontFamily="MEDIUM"
          isUnderlined={false}
          isUppercase={false}
          maxFontSizeMultiplier={1.3}
          size="SMALL"
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Calibre-Medium",
                "fontSize": 27,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#FFFFFF",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "none",
              },
              Object {
                "marginTop": 14,
              },
            ]
          }
        >
          Maybe Later
        </Text>
      </View>
      <Image
        resizeMode="contain"
        source={null}
        style={
          Array [
            Object {
              "height": 328,
              "position": "absolute",
              "right": 56,
              "top": -304,
              "width": 208,
            },
          ]
        }
        top={0}
      />
    </View>
  </Modal>
</RCTSafeAreaView>
`;
